src/webgpu/api/operation/resource_init/texture_zero_init.spec.ts

```diff
@@ -467,127 +423,197 @@ export abstract class TextureZeroInitTest extends GPUTest {
     }
     this.queue.submit([commandEncoder.finish()]);
   }
+}
 
-  static generateParams(readMethods: ReadMethod[]) {
+const paramsBuilder = params()
+  .combine(
+    poptions('readMethod', [
+      ReadMethod.CopyToBuffer,
+      ReadMethod.CopyToTexture,
+      ReadMethod.Sample,
+      ReadMethod.DepthTest,
+      ReadMethod.StencilTest,
+    ])
+  )
+  .combine(poptions('format', kUncompressedTextureFormats))
+  .expand(({ format }) => {
+    const info = kUncompressedTextureFormatInfo[format];
+    let aspects: { aspect: GPUTextureAspect }[] = [];
+    if (info.depth && info.stencil) {
+      aspects = [{ aspect: 'depth-only' }, { aspect: 'stencil-only' }];
+      return aspects;
+    }
+    aspects = [{ aspect: 'all' }];
+    return aspects;
```

Comment:
nit: this could be simplified as below. Alternatively, could merge the `.filter(({ readMethod, aspect }) => ...` logic in here I think.

```suggestion
    if (info.depth && info.stencil) {
      return [{ aspect: 'depth-only' }, { aspect: 'stencil-only' }];
    }
    return [{ aspect: 'all' }];
```
or
```suggestion
    return poptions('aspect', (info.depth && info.stencil) ? ['depth-only', 'stencil-only'] : ['all']);
```